### ISSUE TITLE
MNN 2.9.4版本在运行llm_demo时，Load mnn模型后无法正常继续

### ISSUE THREAD

**Issue by Sodiride123** [alibaba/MNN#3243]
您好，我需要使用MNN 2.9.4实现LLM的推理

拉取的是这个commit 5e93be132e42aed8a41fd2f323fdb78a0b4eb1df

然后发现需要按照2.9.3的Guide实现LLM的推理：
https://mnn-docs.readthedocs.io/en/2.9.3/transformers/llm.html

模型用的就是：Qwen2-0.5B-Instruct

1. 编译没有出问题，正常
`cmake .. -G Ninja DCMAKE_BUILD_TYPE=Release -DMNN_BUILD_SHARED_LIBS=OFF -DMNN_WIN_RUNTIME_MT=ON 
-DMNN_OPENCL=ON -DMNN_BUILD_CONVERTER=ON 
-DMNN_LOW_MEMORY=true -DMNN_CPU_WEIGHT_DEQUANT_GEMM=true -DMNN_BUILD_LLM=true 
-DMNN_SUPPORT_TRANSFORMER_FUSE=true -DMNN_AVX512=true`

2. 执行llm_export.py导出模型，按照提供的命令，结果正常
`python llm_export.py --type Qwen2-0_5B-Instruct --path F:\yy30061803\MNN_LLM\Qwen2-0.5B-Instruct --export --export_token --export_embed --embed_bin --export_mnn`
最后一部分编译日志：
-- Don't has bizCode, use MNNTest for default
-- Start to Convert Other Model Format To MNN Model..., target version: 3
-- [15:19:57] @ 46: ONNX Model ir version: 8
-- [15:19:57] @ 47: ONNX Model opset version: 15
-- Start to Optimize the MNN Net...
-- Save Weight to ./mnn\llm.mnn.weight
-- inputTensors : [ input_ids, position_ids, past_key_values, attention_mask, ]
-- outputTensors: [ logits, presents, ]
-- Converted Success!

    产物也应该是没问题：
    ├── onnx
    |    ├── embeddings_bf16.bin (265,888KB)
    |    ├── llm_config.json (1KB)
    |    ├── llm.onnx (1,930,300KB)
    |    └── tokenizer.txt (3799KB)
    └── mnn
         ├── llm.mnn (592KB)
         └── llm.mnn.weight (245,122KB)

3. 把生成的onnx和mnn里面的产物置于同一个文件夹下，然后自己添加了一个配置文件config.json
4. 尝试运行llm_demo:
`llm_demo.exe F:\yy30061803\MNN_LLM\MNN_2.9.4\transformers\llm\export\Qwen2_0_5B_model_dir\llm.mnn` 
或者 
`llm_demo.exe F:\yy30061803\MNN_LLM\MNN_2.9.4\transformers\llm\export\Qwen2_0_5B_model_dir\llm.mnn prompt.txt`

结果都是：
--config path is F:\yy30061803\MNN_LLM\MNN_2.9.4\transformers\llm\export\Qwen2_0_5B_model_dir\llm.mnn
--The device supports: i8sdot:0, fp16:0, i8mm: 0, sve2: 0
--Can't open file:.tempcache
--Load Cache file error.
--is_single_ = 1
--load tokenizer
--tokenizer_type = 3
--load tokenizer Done
***--load F:\yy30061803\MNN_LLM\MNN_2.9.4\transformers\llm\export\Qwen2_0_5B_model_dir\llm.mnn ...(到这里load mnn没成功，然后就自动退出了)***
--(llm_MNN) F:\yy30061803\MNN_LLM\MNN_2.9.4\build>

**Comment by jxt1234** [comment]
embedding.bin 路径不对。需要在 llm.mnn 同一目录下

**Comment by jxt1234** [comment]
还有 tokenizer.txt 也需要在 llm.mnn 同一目录

**Comment by Sodiride123** [comment]
├── onnx
| ├── embeddings_bf16.bin (265,888KB)
| ├── llm_config.json (1KB)
| ├── llm.onnx (1,930,300KB)
| └── tokenizer.txt (3799KB)
└── mnn
├── llm.mnn (592KB)
└── llm.mnn.weight (245,122KB)

上面这个onnx和mnn是执行llm_export.py之后生成的产物，我已经把他们都放到同一个我自己创建的新目录下面了，也新加了config.json：
├── Qwen2_0_5B_model_dir (自己创建的目录)
| ├── embeddings_bf16.bin (265,888KB)
| ├── llm_config.json (1KB)
| ├── llm.onnx (1,930,300KB)
| ├── tokenizer.txt (3799KB)
| ├── llm.mnn (592KB)
| ├── llm.mnn.weight (245,122KB)
  └── config.json (1KB)

运行llm_demo.exe \Qwen2_0_5B_model_dir\llm.mnn 就会出现之前的状况，tokenizer可以Load成功，Load到mnn就自动退出了。

**Comment by Sodiride123** [comment]
@jxt1234 
之前用的环境是Windows+4090，更改换成Linux+4090又试了一次，还是编译的2.9.4版本https://github.com/alibaba/MNN/commit/5e93be132e42aed8a41fd2f323fdb78a0b4eb1df

这次可以加载mnn成功，不过在交互式对话的时候提示***Unary Op can not execute***：
(LLM_MNN_yy) root@IPMI:~/yy30061803/LLM_MNN/MNN/build# ./llm_demo /root/yy30061803/LLM_MNN/MNN/transformers/llm/export/onnx/llm.mnn
--config path is /root/yy30061803/LLM_MNN/MNN/transformers/llm/export/onnx/llm.mnn
--Can't open file:/sys/devices/system/cpu/cpufreq/schedutil/affected_cpus
--CPU Group: [ 35  10  39  7  29  19  37  5  27  17  20  3  25  15  33  1  23  13  31  21  34  8  38  6  28  18  36  4  26  16  11  2  24  14  32  0  22  12  9  30 ], 1000000 - 2400000
--The device supports: i8sdot:0, fp16:0, i8mm: 0, sve2: 0
--Can't open file:.tempcache
--Load Cache file error.
--### is_single_ = 1
--load tokenizer
--tokenizer_type = 3
--load tokenizer Done
--load /root/yy30061803/LLM_MNN/MNN/transformers/llm/export/onnx/llm.mnn ... Done!
--main, 167, cost time: 1410.145020 ms

--Q: tell me something about computer science

--A: ERROR: Unary Op can not execute
--ERROR: Unary Op can not execute
--Create execution error : 101
--code=2 in onForward, 572 

--Q: 
--A: ERROR: Unary Op can not execute
--ERROR: Unary Op can not execute
--Create execution error : 101
--code=2 in onForward, 572

**Comment by jxt1234** [comment]
估计是你在 windows 上导出，默认用的 pymnn ，这个版本大于等于 3.0.1 .
建议：换成最新代码的 MNN 编译运行。或者在 llmexport.py 时加上 `--mnnconvert ` 指定 MNNConvert.exe 的路径

**Comment by Sodiride123** [comment]
@jxt1234 好的谢谢，感觉Windows上坑比较多，现在换成了Linux环境，代码编译，模型导出全部都是用的2.9.4，然后运行 
./llm_demo.exe llm.mnn

--Can't open file:/sys/devices/system/cpu/cpufreq/schedutil/affected_cpus
--CPU Group: [ 35 10 39 7 29 19 37 5 27 17 20 3 25 15 33 1 23 13 31 21 34 8 38 6 28 18 36 4 26 16 11 2 24 14 32 0 22 12 9 30 ], 1000000 - 2400000
--The device supports: i8sdot:0, fp16:0, i8mm: 0, sve2: 0
--Can't open file:.tempcache
--Load Cache file error.
--### is_single_ = 1
--load tokenizer
--tokenizer_type = 3
--load tokenizer Done
--load /root/yy30061803/LLM_MNN/MNN/transformers/llm/export/onnx/llm.mnn ... Done!
--main, 167, cost time: 1410.145020 ms

--Q: tell me something about computer science

--A: ERROR: Unary Op can not execute
--ERROR: Unary Op can not execute
--Create execution error : 101
--code=2 in onForward, 572

是不是不支持Unary算子，只能用 >2.9.4的版本编译？

**Comment by jxt1234** [comment]
这个是 3.0.1 加入的 silu 算子（归属 unary 大类）。必须用 3.0.1 之后的版本编译。

你应该是没有编 MNNConvert ，非要用 2.9.4 的话，本地 build 目录下执行 cmake ..  -DMNN_BUILD_CONVERT=ON && build，再重新 export 。

**Comment by andyt9527** [comment]
tempcache这个是什么文件？

--Can't open file:.tempcache
--Load Cache file error.

这两行log会有问题么
